JASP-Engine/JASP/R/correlationbayesian.R

```diff
@@ -15,2850 +15,1312 @@
-						}
-
-						if (isTRUE(credibleInterval)) {
-						    upperCiList[[length(upperCiList)+1]] <- .clean(reportUpperCi)
-						    lowerCiList[[length(lowerCiList)+1]] <- .clean(reportLowerCi)
-						}
-					} # Close: Results reporting from 1 retrieved or 2 performed
-
-					# Note: No retrieval and no performance. Thus fill in full stops
-					#
-					if (retrievalFailure && perform!="run") {
-						# No data retrieved from state and nothing run
-						row[[length(row)+1]] <- "."
-						bayesFactorsList[[length(bayesFactorsList)+1]] <- "."
-						upperCiList[[length(upperCiList)+1]] <- "."
-						lowerCiList[[length(lowerCiList)+1]] <- "."
-					} # Close: Results reporting from initialisation
-				} # Close loop: each column
-				#
-				# Hence, we now have all the information to fill in the rest of the table
-
-				# Reporting: Take each bf in the list i+1, i+2, .. numberOfVariables and fill in
-				#
-				if (isTRUE(reportBayesFactors)) {
-					for (bf in bayesFactorsList) {
-						row[[length(row)+1]] <- bf
-					}
-				}
-
-				if (isTRUE(credibleInterval)) {
-				    for (upperCi in upperCiList) {
-				        row[[length(row)+1]] <- upperCi
-				    }
-
-				    for (lowerCi in lowerCiList) {
-				        row[[length(row)+1]] <- lowerCi
-				    }
-				}
-			} # Close loop: each test
-
-			# MarkUp: Per row (variable) Design
-			names(row) <- columnNames
-			row[[".variable"]] <- variableName
-			if (length(rowFootnotes) > 0)
-				row[[".footnotes"]] <- rowFootnotes
-			allRows[[i]] <- row
-		} # Note: close what goes each row
-	} # Close: the naming and initialisation of structure of each row
-
-	# Give the column names to
-	schema <- list(fields=fields)
-	correlationTable[["schema"]] <- schema
-	correlationTable[["data"]] <- allRows
-	correlationTable[["footnotes"]] <- as.list(footnotes)
-
-	return(list(correlationTable=correlationTable,
-				test=tests,
-				variables=variables,
-				rows=allRows,
-				memoryExcludePairwise=memoryExcludePairwise,
-				memoryExcludeListwise=memoryExcludeListwise))
-}
-## Help functions ------------------------------------------------------------
-# 0.1 Prior specification Pearson's Rho
-.excludePairwiseCorData <- function(v1, v2) {
-	# To exclude the data pairwise
-	#
-	screenedData <- list(v1=v1, v2=v2)
-
-	removeIndex1 <- which(is.na(v1))
-	removeIndex2 <- which(is.na(v2))
-	removeIndex <- unique(c(removeIndex1, removeIndex2))
-	if (length(removeIndex) > 0) {
-		screenedData$v1 <- v1[-(removeIndex)]
-		screenedData$v2 <- v2[-(removeIndex)]
-	}
-
-	return(screenedData)
-}
+  # 2. Data retrieval  --------
+  #
+  if (readyMatrix | readyPairs) {
+    if (is.null(dataset)) 
+      dataset <- .corBayesReadData(dataset, options)
+  }
 
-.stretchedBeta <- function(rho, alpha, beta) {
-	result <- 1/2*dbeta((rho+1)/2, alpha, beta)
-	return(result)
+  # 3. Table: Get, compute and fill -----------
+  #
+  allBfObjects <- .makeTableCorBayes("jaspResults"=jaspResults, "allBfObjects"=allBfObjects,
```

Comment:
It feels unnatural that `makeTableCorBayes` doesn't return.. a table?

Is there a reason you're creating one massive state object? Because you don't need to do that anymore (you can create as many state objects as you'd like)... It's great because then you can compartmentalize your code and don't couple everything so tightly.